cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/JbodStorageMockTest.java

```diff
@@ -132,12 +133,13 @@ private void init() {
         mockKube.start();
 
         PlatformFeaturesAvailability pfa = new PlatformFeaturesAvailability(false, KubernetesVersion.V1_16);
+        KubernetesRestartEventPublisher restartEventPublisher = KubernetesRestartEventPublisher.createPublisher(client, "op", pfa.hasEventsApiV1());
         // creating the Kafka operator
         ResourceOperatorSupplier ros =
                 new ResourceOperatorSupplier(this.vertx, this.client,
                         ResourceUtils.zookeeperLeaderFinder(this.vertx, this.client),
                         ResourceUtils.adminClientProvider(), ResourceUtils.zookeeperScalerProvider(),
-                        ResourceUtils.metricsProvider(), pfa, 60_000L);
+                        ResourceUtils.metricsProvider(), pfa, 60_000L, restartEventPublisher);
```

Comment:
Can't use the other constructor as the test fails, so have added a mock instead.

Reply:
Why does it fail? When the Event Publisher is created inside the Resource OPerator Supplier it should create exactly the same as you create here, or?

Reply:
```
        // creating the Kafka operator
//        ResourceOperatorSupplier ros =
//                new ResourceOperatorSupplier(this.vertx, this.client,
//                        ResourceUtils.zookeeperLeaderFinder(this.vertx, this.client),
//                        ResourceUtils.adminClientProvider(), ResourceUtils.zookeeperScalerProvider(),
//                        ResourceUtils.metricsProvider(), pfa, 60_000L, mock(KubernetesRestartEventPublisher.class));

        ResourceOperatorSupplier ros = new ResourceOperatorSupplier(vertx, client, pfa, 10_000, "op");
```

Test run:

```
LF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/liam/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.17.2/log4j-slf4j-impl-2.17.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/liam/.m2/repository/org/slf4j/slf4j-reload4j/1.7.36/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Jun 29, 2022 9:18:21 PM okhttp3.mockwebserver.MockWebServer$3 execute
INFO: MockWebServer[60875] starting to accept connections
Jun 29, 2022 9:18:21 PM okhttp3.mockwebserver.MockWebServer$3 execute
INFO: MockWebServer[35145] starting to accept connections
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
2022-06-29 21:18:22 WARN  VersionUsageUtils:60 - The client is using resource type 'kafkas' with unstable version 'v1beta2'
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: POST /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
2022-06-29 21:18:22 WARN  VersionUsageUtils:60 - The client is using resource type 'strimzipodsets' with unstable version 'v1beta2'
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka HTTP/1.1 and responded: HTTP/1.1 200 OK
2022-06-29 21:18:22 INFO  StrimziPodSetController:440 - Starting the StrimziPodSet controller
2022-06-29 21:18:22 INFO  StrimziPodSetController:408 - Starting StrimziPodSet controller for namespace test-jbod-storage
2022-06-29 21:18:22 INFO  StrimziPodSetController:410 - Waiting for informers to sync
2022-06-29 21:18:22 INFO  StrimziPodSetController:415 - Informers are in-sync
2022-06-29 21:18:22 INFO  StrimziPodSetController:448 - Requesting the StrimziPodSet controller to stop
2022-06-29 21:18:22 INFO  StrimziPodSetController:429 - Stopping StrimziPodSet controller
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$3 acceptConnections
INFO: MockWebServer[35145] done accepting connections: Socket closed
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /apis/apps/v1/statefulsets?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /apis/apps/v1/deployments?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /api/v1/services?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka&resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[35145] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols

java.lang.NullPointerException
	at io.micrometer.core.instrument.Counter$Builder.register(Counter.java:128)
	at io.strimzi.operator.common.MicrometerMetricsProvider.counter(MicrometerMetricsProvider.java:61)
	at io.strimzi.operator.common.Operator.lambda$getCounter$2(Operator.java:130)
	at io.strimzi.operator.common.Operator.lambda$metric$1(Operator.java:126)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.strimzi.operator.common.Operator.metric(Operator.java:126)
	at io.strimzi.operator.common.Operator.getCounter(Operator.java:130)
	at io.strimzi.operator.common.AbstractOperator.reconciliationsCounter(AbstractOperator.java:544)
	at io.strimzi.operator.common.AbstractOperator.reconcile(AbstractOperator.java:167)
	at io.strimzi.operator.cluster.operator.assembly.JbodStorageMockTest.testReconcileWithUpdateVolumeIdJbod(JbodStorageMockTest.java:294)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:155)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$3 execute
INFO: MockWebServer[41795] starting to accept connections
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: POST /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka HTTP/1.1 and responded: HTTP/1.1 200 OK
2022-06-29 21:18:22 INFO  StrimziPodSetController:440 - Starting the StrimziPodSet controller
2022-06-29 21:18:22 INFO  StrimziPodSetController:408 - Starting StrimziPodSet controller for namespace test-jbod-storage
2022-06-29 21:18:22 INFO  StrimziPodSetController:410 - Waiting for informers to sync
2022-06-29 21:18:22 INFO  StrimziPodSetController:415 - Informers are in-sync
2022-06-29 21:18:22 INFO  StrimziPodSetController:448 - Requesting the StrimziPodSet controller to stop
2022-06-29 21:18:22 INFO  StrimziPodSetController:429 - Stopping StrimziPodSet controller
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$3 acceptConnections
INFO: MockWebServer[41795] done accepting connections: Socket closed
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /api/v1/services?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka&resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /apis/apps/v1/deployments?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41795] received request: GET /apis/apps/v1/statefulsets?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols

java.lang.NullPointerException
	at io.micrometer.core.instrument.Counter$Builder.register(Counter.java:128)
	at io.strimzi.operator.common.MicrometerMetricsProvider.counter(MicrometerMetricsProvider.java:61)
	at io.strimzi.operator.common.Operator.lambda$getCounter$2(Operator.java:130)
	at io.strimzi.operator.common.Operator.lambda$metric$1(Operator.java:126)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.strimzi.operator.common.Operator.metric(Operator.java:126)
	at io.strimzi.operator.common.Operator.getCounter(Operator.java:130)
	at io.strimzi.operator.common.AbstractOperator.reconciliationsCounter(AbstractOperator.java:544)
	at io.strimzi.operator.common.AbstractOperator.reconcile(AbstractOperator.java:167)
	at io.strimzi.operator.cluster.operator.assembly.JbodStorageMockTest.testReconcileWithNewVolumeAddedToJbodStorage(JbodStorageMockTest.java:215)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:155)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

Jun 29, 2022 9:18:22 PM okhttp3.mockwebserver.MockWebServer$3 execute
INFO: MockWebServer[41321] starting to accept connections
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: POST /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka HTTP/1.1 and responded: HTTP/1.1 200 OK
2022-06-29 21:18:23 INFO  StrimziPodSetController:440 - Starting the StrimziPodSet controller
2022-06-29 21:18:23 INFO  StrimziPodSetController:408 - Starting StrimziPodSet controller for namespace test-jbod-storage
2022-06-29 21:18:23 INFO  StrimziPodSetController:410 - Waiting for informers to sync
2022-06-29 21:18:23 INFO  StrimziPodSetController:415 - Informers are in-sync
2022-06-29 21:18:23 INFO  StrimziPodSetController:448 - Requesting the StrimziPodSet controller to stop
2022-06-29 21:18:23 INFO  StrimziPodSetController:429 - Stopping StrimziPodSet controller
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$3 acceptConnections
INFO: MockWebServer[41321] done accepting connections: Socket closed
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /api/v1/services?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /apis/apps/v1/deployments?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /apis/apps/v1/statefulsets?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[41321] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka&resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols

java.lang.NullPointerException
	at io.micrometer.core.instrument.Counter$Builder.register(Counter.java:128)
	at io.strimzi.operator.common.MicrometerMetricsProvider.counter(MicrometerMetricsProvider.java:61)
	at io.strimzi.operator.common.Operator.lambda$getCounter$2(Operator.java:130)
	at io.strimzi.operator.common.Operator.lambda$metric$1(Operator.java:126)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.strimzi.operator.common.Operator.metric(Operator.java:126)
	at io.strimzi.operator.common.Operator.getCounter(Operator.java:130)
	at io.strimzi.operator.common.AbstractOperator.reconciliationsCounter(AbstractOperator.java:544)
	at io.strimzi.operator.common.AbstractOperator.reconcile(AbstractOperator.java:167)
	at io.strimzi.operator.cluster.operator.assembly.JbodStorageMockTest.testReconcileWithVolumeRemovedFromJbodStorage(JbodStorageMockTest.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:155)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$3 execute
INFO: MockWebServer[43351] starting to accept connections
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: POST /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: POST /apis/apiextensions.k8s.io/v1/customresourcedefinitions HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets HTTP/1.1 and responded: HTTP/1.1 200 OK
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka HTTP/1.1 and responded: HTTP/1.1 200 OK
2022-06-29 21:18:23 INFO  StrimziPodSetController:440 - Starting the StrimziPodSet controller
2022-06-29 21:18:23 INFO  StrimziPodSetController:408 - Starting StrimziPodSet controller for namespace test-jbod-storage
2022-06-29 21:18:23 INFO  StrimziPodSetController:410 - Waiting for informers to sync
2022-06-29 21:18:23 INFO  StrimziPodSetController:415 - Informers are in-sync
2022-06-29 21:18:23 INFO  StrimziPodSetController:448 - Requesting the StrimziPodSet controller to stop
2022-06-29 21:18:23 INFO  StrimziPodSetController:429 - Stopping StrimziPodSet controller
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$3 acceptConnections
INFO: MockWebServer[43351] done accepting connections: Socket closed
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /apis/core.strimzi.io/v1beta2/namespaces/test-jbod-storage/strimzipodsets?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /api/v1/namespaces/test-jbod-storage/pods?labelSelector=strimzi.io%2Fkind%3DKafka&resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /apis/apps/v1/deployments?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /apis/kafka.strimzi.io/v1beta2/namespaces/test-jbod-storage/kafkas?resourceVersion=3&allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /apis/apps/v1/statefulsets?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /api/v1/services?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols
Jun 29, 2022 9:18:23 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[43351] received request: GET /api/v1/pods?allowWatchBookmarks=true&watch=true HTTP/1.1 and responded: HTTP/1.1 101 Switching Protocols

java.lang.NullPointerException
	at io.micrometer.core.instrument.Counter$Builder.register(Counter.java:128)
	at io.strimzi.operator.common.MicrometerMetricsProvider.counter(MicrometerMetricsProvider.java:61)
	at io.strimzi.operator.common.Operator.lambda$getCounter$2(Operator.java:130)
	at io.strimzi.operator.common.Operator.lambda$metric$1(Operator.java:126)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.strimzi.operator.common.Operator.metric(Operator.java:126)
	at io.strimzi.operator.common.Operator.getCounter(Operator.java:130)
	at io.strimzi.operator.common.AbstractOperator.reconciliationsCounter(AbstractOperator.java:544)
	at io.strimzi.operator.common.AbstractOperator.reconcile(AbstractOperator.java:167)
	at io.strimzi.operator.cluster.operator.assembly.JbodStorageMockTest.testJbodStorageCreatesPersistentVolumeClaimsMatchingKafkaVolumes(JbodStorageMockTest.java:162)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:155)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

Reply:
I'm not saying the metric provider is not needed. The EventPublisher is not needed.

Reply:
I feel like a whole new constructor to avoid one more dependency is getting silly, @scholzj, and tbh mate, I'm getting a bit frustrated. 

1. You suggested I add the `KubernetesRestartEventPublisher` to `ResourceOperatorSupplier`, instead of as a separate dependency to classes needing it, I did so.
2. You suggested that I not mock it unless needed, I did so.
3. You then wondered why I was creating an instance of `KubernetesRestartEventPublisher`to pass to the `ResourceOperatorSupplier`, and suggested I use the alternate constructor. 
4. I have changed it to use the alternate constructor where that works

I've responded to your feedback throughout this PR, but you keep shifting the goalposts. 

I'm not going to create yet another overloaded constructor to reduce "noise", because the "noise" you refer to is an additional parameter. And IMO the multitude of overloaded constructors that already exist in the Strimzi codebase is far more noisy than this.

Reply:
> I feel like a whole new constructor to avoid one more parameter is getting silly, @scholzj, and tbh mate, I'm getting a bit frustrated.
>
> 1. You suggested I add the `KubernetesRestartEventPublisher` to `ResourceOperatorSupplier`, instead of as a separate dependency to classes needing it, I did so.
> 2. You suggested that I not mock it unless needed, I did so.
> 3. You then wondered why I was creating an instance of `KubernetesRestartEventPublisher`to pass to the `ResourceOperatorSupplier`, and suggested I use the alternate constructor.
> 4. I have changed it to use the alternate constructor where that works

While you decided to write it a little bit differently, the Event publisher is essentially on the same level as an EventOperator would be and if we one day need to use events also for something more, it will be likely replaced with it. And as so yes, I think it belongs with the other operators into the `ResourceOperatorSupplier` class and should not be passed separately.

Yes, I'm trying to avoid PRs doing unnecessary changes which:
* Make it harder to trace through the history
* Make it harder to maintain the code

You decided to add `KubernetesRestartEventPublisher` as a parameter to the existing constructor. You seem to use it in 8 places => all of these places are in test. And all of them seem to just create the `KubernetesRestartEventPublisher` either as a proper object or a mock and pass it to the constructor without doing anything else with it. I struggle to see any positive value of that when you do not use it, do not mock any methods, nothing. So why is it needed?

When you did bunch of unrelated changes where you just applied some IDE recommendations on a completely unrelated place or made unnecessary changes to whitespaces, I asked for the reason where it was not obvious. But I did not asked you to revert them in many cases. That actually makes my review harder, but I decided that I can live with it in order to get this finally merged. But here I'm trying to find out you why do we need a new parameter in a constructor which does not seem to be needed on a first look. I think that is a valid comment regardless how many times you changed it or how long the PR is opened.